Core/src/Surfaces/LineSurface.cpp

```diff
@@ -77,6 +78,12 @@ Acts::Result<Acts::Vector2> Acts::LineSurface::globalToLocal(
   // assign the right sign
   double sign = ((lineDirection.cross(momentum)).dot(decVec) < 0.) ? -1. : 1.;
   lposition[eBoundLoc0] *= sign;
+
+  if ((localToGlobal(gctx, lposition, momentum) - position).norm() >
```

Comment:
I think  `localToGlobal` would be expensive

I suggest to use the boundaries defined `LineBounds.hpp`

`std::abs(lposition[0) < r && std::abs(lposition[1]) < halfz`

Reply:
`localToGlobal` will not only check the bounds but also how far we got projected onto the surface by the code above. the calculation above will always produce a local position even if we are not on the surface

Reply:
> the calculation above will always produce a local position even if we are not on the surface

This is true but you can still use `lposition[0]` and `lposition[1]` for boundary check regardless whether it is on the surface or not.

Reply:
that is true but the `lposition` you get will not represent the global `position` if you provide input that is not in the PCA. it will project you onto the surface and I don't think it should

`globalToLocal(<random global position>)` will always give you an answer and the inverse `localToGlobal(globalToLocal(<random global position>)) != random global position` will not give you the same position back because you potentially moved it across you whole detector

Reply:
oh I understand the point. In that case you can check if the track direction (`t`) is orthogonal to the radial vector (`w`)

```c++
  // direction
  vector3  t = normalize(mom);
  vector3 r = position;
 
  // line center
  c = sCenter 
 
  // line axis direction
  vector3 l = lineDirection
  
  // Projection of (track position - center) to the line 
  double A = dot(r-c,l)
  
  // Radial vector: Vector orthogonal to the line and passing the track position
  // w = r - (c + ((r - c) * l)l)
  const vector3 w = r - (c + A * l);

  if ( std::abs(dot( t* w)) > tolerance){
     // failure
  } 
```

It is also expensive but cheaper that `localToGlobal` I suppose.

Reply:
but I think `localToGlobal` does exactly that, no? at least when I look at the implementation is seems quite simple

Reply:
I thought this line would be expensive.
```c++
Vector3 locZinGlobal = sTransform * Vector3(0., 0., lposition[eBoundLoc1]);
```

Reply:
Tbh, I'm also slightly worried about the performance implications.

Reply:
I don't feel like this is a particular hot path but I am happy to improve performance on suggestions. right now I just want to fix the bug and face a ton of downstream failures

for example we have another problem with numerical instabilities. some intersections will give you global positions which are not on surface according to these changes